client/python/tests/unit/test_event.py

```diff
@@ -58,3 +89,43 @@ def test_event_class(name, event_type):
     assert test_event.message.job_id == "job-1"
     assert test_event.message.job_set_id == "job-set-1"
     assert test_event.message.queue == "queue-1"
+
+
+@pytest.mark.parametrize(
+    "name",
+    [
+        "submitted",
+        "queued",
+        "duplicate_found",
+        "leased",
+        "lease_returned",
+        "lease_expired",
+        "pending",
+        "running",
+        "unable_to_schedule",
+        "failed",
+        "succeeded",
+        "reprioritized",
+        "cancelling",
+        "cancelled",
+        "terminated",
+        "utilisation",
+        "ingress_info",
+        "reprioritizing",
+        "updated",
+    ],
+)
+def test_unmarshal_event_response(name):
+    test_event = FakeEventStreamMessage(name)
+    test_event_from_class = Event(test_event)
+
+    test_event_from_method = tester.unmarshal_event_response(test_event)
```

Comment:
This leaves a bad taste in my mouth.  I really don't think we need all this setup to test a static function that takes an EventMessage.  

We aren't testing any RPC calls here so I'd like to see a way to avoid having to create mock grpc servers to test this.

Reply:
Maybe you can move this to the event class?

Reply:
Yeah I was not a huge fan of this either, but I did not experiment with trying to use the ArmadaClass Directly. I will see if I can get this to work and go from there.